mapviz_plugins/src/string_plugin.cpp

```diff
@@ -387,9 +387,19 @@ namespace mapviz_plugins
     offset_y_ = offset;
   }
 
-  void StringPlugin::stringCallback(const std_msgs::StringConstPtr& str)
+#define IS_INSTANCE(msg, type) \
+  (msg->getDataType() == ros::message_traits::datatype<type>())
```

Comment:
You should probably check with @pjreed if he wants this as a macro

Reply:
While this isn't the end of the world since it's at least not in a header, but is there any reason for this to not be a function?

Reply:
I guess not. It could be a templated function. My excuse is everyone else is doing exactly this.